OpenDreamRuntime/Objects/Types/DreamObjectSavefile.cs

```diff
@@ -150,31 +164,372 @@ public sealed class SavefileDirectory : Dictionary<string, DreamValue> { }
+                        else
+                            l.AddValue(DeserializeJsonValue(SFDreamListValue.AssocKeys[i]));
+                    }
+                }
+                return new DreamValue(l);
+            case SFDreamObjectPathValue SFDreamObjectPath:
+                SFDreamJsonValue storedObjectVars = SFDreamObjectPath;
+                SFDreamJsonValue searchDir = SFDreamObjectPath;
+                while(searchDir != _rootNode)
+                    if(!searchDir.TryGetValue(SFDreamObjectPath.Path, out storedObjectVars!))
+                        searchDir = SeekTo("..");
+                    else
+                        break;
+
+                if(storedObjectVars!.TryGetValue("type", out SFDreamJsonValue? storedObjectTypeJSON) && DeserializeJsonValue(storedObjectTypeJSON).TryGetValueAsType(out TreeEntry? objecTypeActual)) {
+                    DreamObject resultObj = _objectTree.CreateObject(objecTypeActual);
+                    foreach(string key in storedObjectVars.Keys){
+                        if(key == "type")
+                            continue;
+                        resultObj.SetVariable(key, DeserializeJsonValue(storedObjectVars[key]));
+                    }
+                    resultObj.InitSpawn(new DreamProcArguments());
+                    resultObj.SpawnProc("Read", null, [new DreamValue(this)]);
+                    return new DreamValue(resultObj);
+                } else
+                    throw new InvalidDataException("Unable to deserialize object in savefile: " + ((storedObjectTypeJSON as SFDreamType) is null ? "no type specified (corrupted savefile?)" : "invalid type "+((SFDreamType)storedObjectTypeJSON!).TypePath));
+            case SFDreamType SFDreamTypeValue:
+                if(_objectTree.TryGetTreeEntry(SFDreamTypeValue.TypePath, out var type)) {
+                    return new DreamValue(type);
+                } else {
+                    return DreamValue.Null;
+                }
+            case SFDreamPrimitive SFDreamPrimitive:
+                return SFDreamPrimitive.Value;
+        }
+        return DreamValue.Null;
+    }
+
+    /// <summary>
+    /// Serialize byond values/objects into savefile data
+    /// </summary>
+    public SFDreamJsonValue SerializeDreamValue(DreamValue val, int objectCount = 0) {
+        switch (val.Type) {
+            case DreamValue.DreamValueType.String:
+            case DreamValue.DreamValueType.Float:
+                return new SFDreamPrimitive { Value = val };
+            case DreamValue.DreamValueType.DreamType:
+                return new SFDreamType { TypePath = val.MustGetValueAsType().Path };
+            case DreamValue.DreamValueType.DreamResource:
+                var dreamResource = val.MustGetValueAsDreamResource();
+                return new SFDreamFileValue {
+                    Length = dreamResource.ResourceData!.Length,
+                    //Crc32 =  new System.IO.Hashing.Crc32().
```

Comment:
https://www.nuget.org/packages/System.IO.Hashing for crc, importtext might not align with the crc if its from byond which we could ignore

Reply:
Rolled my own, easier than importing a whole package.